modules/signals/src/with-hooks.ts

```diff
@@ -16,11 +16,45 @@ type HooksFactory<Input extends SignalStoreFeatureResult> = (
   >
 ) => void;
 
+type HooksFactory<Input extends SignalStoreFeatureResult> = (
+  store: Prettify<
+    SignalStoreSlices<Input['state']> &
+      Input['signals'] &
+      Input['methods'] &
+      StateSignal<Prettify<Input['state']>>
+  >
+) => {
+  onInit?: () => void;
+  onDestroy?: () => void;
+};
+
 export function withHooks<Input extends SignalStoreFeatureResult>(hooks: {
-  onInit?: HooksFactory<Input>;
-  onDestroy?: HooksFactory<Input>;
-}): SignalStoreFeature<Input, EmptyFeatureResult> {
+  onInit?: HookFn<Input>;
+  onDestroy?: HookFn<Input>;
+}): SignalStoreFeature<Input, EmptyFeatureResult>;
+export function withHooks<Input extends SignalStoreFeatureResult>(
+  hooks: HooksFactory<Input>
+): SignalStoreFeature<Input, EmptyFeatureResult>;
+
+export function withHooks<Input extends SignalStoreFeatureResult>(
+  hooksOrFactory:
+    | {
+        onInit?: HookFn<Input>;
+        onDestroy?: HookFn<Input>;
+      }
+    | HooksFactory<Input>
+): SignalStoreFeature<Input, EmptyFeatureResult> {
   return (store) => {
+    const storeProps = {
```

Comment:
to avoid redundant code, `storeProps` can be used on line 68

Reply:
Sorry, missed that one. Should be all good now.